bot/exts/utils/utils.py

```diff
@@ -257,7 +243,70 @@ async def send_pep_zero(self, ctx: Context) -> None:
         pep_embed.add_field(name="Created", value="13-Jul-2000")
         pep_embed.add_field(name="Type", value="Informational")
 
-        await ctx.send(embed=pep_embed)
+        return pep_embed
+
+    async def validate_pep_number(self, ctx: Context, pep_nr: int) -> bool:
+        """Validate is PEP number valid. When it isn't, send error and return False. Otherwise return True."""
+        if (
+            pep_nr not in self.peps
+            and (self.last_refreshed_peps + timedelta(minutes=30)) <= datetime.now()
+            and len(str(pep_nr)) < 5
+        ):
+            await self.refresh_peps_urls()
+
+        if pep_nr not in self.peps:
+            log.trace(f"PEP {pep_nr} was not found")
+            not_found = f"PEP {pep_nr} does not exist."
+            await self.send_pep_error_embed(ctx, "PEP not found", not_found)
+            return False
+
+        return True
+
+    def generate_pep_embed(self, pep_header: Dict, pep_nr: int) -> Embed:
+        """Generate PEP embed based on PEP headers data."""
+        # Assemble the embed
+        pep_embed = Embed(
+            title=f"**PEP {pep_nr} - {pep_header['Title']}**",
+            description=f"[Link]({self.BASE_PEP_URL}{pep_nr:04})",
+        )
+
+        pep_embed.set_thumbnail(url=ICON_URL)
+
+        # Add the interesting information
+        fields_to_check = ("Status", "Python-Version", "Created", "Type")
+        for field in fields_to_check:
+            # Check for a PEP metadata field that is present but has an empty value
+            # embed field values can't contain an empty string
+            if pep_header.get(field, ""):
+                pep_embed.add_field(name=field, value=pep_header[field])
+
+        return pep_embed
+
+    @pep_cache(arg_offset=2)
+    async def get_pep_embed(self, ctx: Context, pep_nr: int) -> Optional[Embed]:
+        """Fetch, generate and return PEP embed. When any error occur, use `self.send_pep_error_embed`."""
+        response = await self.bot.http_session.get(self.peps[pep_nr])
```

Comment:
We shouldn't need the error handling code below if we raise for status here and let it propagate to the user

Reply:
You mean setting raise_for_status to `True` and when this send any error?

Reply:
Ah nevermind that, this shouldn't go directly to the user since we don't have the proper handling for it in the error handler.
Though raising for status and logging the exception (as a warning or an exception) would cover all the status errors along with things like connection errors which are currently unhandled

Reply:
I think it's not necessary, as this already logs response status code and I think connection error will raise error anyway?